src/components/tasks/card/index.tsx

```diff
@@ -39,33 +39,36 @@ const Card: FC<Props> = ({
   const [showEditButton, setShowEditButton] = useState(false);
   const [keyLongPressed] = useKeyLongPressed();
   const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const context = useAppContext();
+  const { actions, state } = context || {};
+
   useEffect(() => {
     const isAltKeyLongPressed = keyLongPressed === ALT_KEY;
     if (isAltKeyLongPressed) {
       setShowEditButton(true)
     }
   }, [keyLongPressed]);
   useEffect(() => {
-    (async () => {
-      try{
-        const { requestPromise } = fetch(
-          { 
-            url: `${ITEM_BY_FILTER_URL}`,
-            params: {
-              itemType: 'TASK',
-              itemId: `${cardDetails.id}`
-            }
-          });
-        const { data: result } = await requestPromise
-        setTaskTagLevel(result.data)
-       } catch (err: any) {
-         toast(ERROR, err.message);
-       }
-    })()
-  },[])
-  
-  const context = useAppContext() ;
-  const { actions } = context || {}
+    if (state.isLoggedIn) {
+      (async () => {
+        try {
+          const { requestPromise } = fetch(
+            {
+              url: `${ITEM_BY_FILTER_URL}`,
+              params: {
+                itemType: 'TASK',
+                itemId: `${cardDetails.id}`
+              }
+            });
```

Comment:
In Strict mode fetching data without a cleanup function to ignore the request might cause bugs in development mode. 
Attaching a link to the react beta docs where they give a good example of how to fetch data. 

[https://beta.reactjs.org/learn/synchronizing-with-effects#fetching-data](url)

Reply:
I haven't added this. I have created a refactor ticket for this https://github.com/Real-Dev-Squad/website-status/issues/406 
This component needs a revamp.